pubsub/topics/topics_test.go

```diff
@@ -293,3 +295,26 @@ func TestDelete(t *testing.T) {
 		t.Fatalf("got topic = %q; want none", topicID)
 	}
 }
+
+func TestTopicKinesis(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	buf := new(bytes.Buffer)
+
+	// Use the pstest fake with emulator settings since Pub/Sub service expects real AWS Kinesis
+	// resources, which we cannot provide in a samples test.
+	srv := pstest.NewServer()
+	os.Setenv("PUBSUB_EMULATOR_HOST", srv.Addr)
```

Comment:
I believe this permanently sets the env variable which may affect other tests' execution. Recommend using https://pkg.go.dev/testing#T.Setenv instead.